tools/autograph-monitor/monitor.go

```diff
@@ -153,7 +153,7 @@ func monitor() (err error) {
 			err = verifyContentSignature(response)
 		case contentsignaturepki.Type:
 			log.Printf("Verifying content signature pki from signer %q", response.SignerID)
-			err = contentsignaturepki.Verify(response.X5U, response.Signature, []byte(inputdata))
+			err = verifyContentSignature(response)
```

Comment:
Apologies, as I'm new to reviewing autograph code, but what is the implication of this change?

Reply:
Thanks for taking a look AJ!

The `contentsignaturepki.Verify` from `signer/contentsignaturepki` https://github.com/mozilla-services/autograph/blob/master/signer/contentsignaturepki/contentsignature.go#L321 just verifies the leaf or EE in the root -> intermediate -> EE chain.

The `verifyContentSignature` from `tools/autograph-monitor` https://github.com/mozilla-services/autograph/blob/d58a6b57fb2e7f6f72cd29496e0d25080b040a70/tools/autograph-monitor/contentsignaturepki.go#L33 verifies the response data and uses [verifyCertChain](https://github.com/mozilla-services/autograph/blob/d58a6b57fb2e7f6f72cd29496e0d25080b040a70/tools/autograph-monitor/contentsignaturepki.go#L88) to check the full root -> intermediate -> EE chain. It sends a soft SNS notification if any cert has <30 days until it expires and errors if any cert is <15 days to expiration or the cert isn't currently valid (i.e. cert NotBefore < now).

There's also a third chain verification implementation in the code that uploads the chain to S3, I filed https://github.com/mozilla-services/autograph/issues/593 to clean them up.

Reply:
Ok awesome! Thank you for the explanation, this sounds great to me.